packages/legend-application-query/src/stores/data-space/DataSpaceTemplateQueryCreatorStore.ts

```diff
@@ -90,62 +104,188 @@ export class DataSpaceTemplateQueryCreatorStore extends QueryEditorStore {
     };
   }
 
+  override *buildGraph(): GeneratorFn<void> {
+    // do nothing
+  }
+
   async initializeQueryBuilderState(): Promise<QueryBuilderState> {
+    let dataSpaceAnalysisResult;
+    let buildFullGraph = false;
+    const supportBuildMinimalGraph =
+      this.applicationStore.config.options.TEMPORARY__enableMinimalGraph;
+    if (
+      this.enableMinialGraphForDataSpaceLoadingPerformance &&
+      supportBuildMinimalGraph
+    ) {
+      try {
+        const stopWatch = new StopWatch();
+        const project = StoreProjectData.serialization.fromJson(
+          await this.depotServerClient.getProject(
+            this.groupId,
+            this.artifactId,
+          ),
+        );
+        this.initState.setMessage('Fetching dataspace analysis result...');
+        // initialize system
+        stopWatch.record();
+        await this.graphManagerState.initializeSystem();
+        stopWatch.record(GRAPH_MANAGER_EVENT.INITIALIZE_GRAPH_SYSTEM__SUCCESS);
+
+        const graph_buildReport = createGraphBuilderReport();
+        const dependency_buildReport = createGraphBuilderReport();
+        dataSpaceAnalysisResult = await DSL_DataSpace_getGraphManagerExtension(
+          this.graphManagerState.graphManager,
+        ).analyzeDataSpaceCoverage(
+          this.dataSpacePath,
+          () =>
+            retrieveProjectEntitiesWithDependencies(
+              project,
+              this.versionId,
+              this.depotServerClient,
+            ),
+          () =>
+            retrieveAnalyticsResultCache(
+              project,
+              this.versionId,
+              this.dataSpacePath,
+              this.depotServerClient,
+            ),
+          undefined,
+          graph_buildReport,
+          this.graphManagerState.graph,
+          this.executionContext,
+          undefined,
+          this.getProjectInfo(),
+          this.applicationStore.notificationService,
+          this.depotServerClient,
+        );
+        const mappingPath = dataSpaceAnalysisResult.executionContextsIndex.get(
+          this.executionContext,
+        )?.mapping;
+        if (mappingPath) {
+          // report
+          stopWatch.record(GRAPH_MANAGER_EVENT.INITIALIZE_GRAPH__SUCCESS);
+          const graphBuilderReportData = {
+            timings:
+              this.applicationStore.timeService.finalizeTimingsRecord(
+                stopWatch,
+              ),
+            dependencies: dependency_buildReport,
+            dependenciesCount:
+              this.graphManagerState.graph.dependencyManager
+                .numberOfDependencies,
+            graph: graph_buildReport,
+          };
+          this.applicationStore.logService.info(
+            LogEvent.create(GRAPH_MANAGER_EVENT.INITIALIZE_GRAPH__SUCCESS),
+            graphBuilderReportData,
+          );
+        } else {
+          this.graphManagerState.graph =
```

Comment:
any reason this else is needed, dont we already build the full graph if there is an error below ?